src/client.cpp

```diff
@@ -0,0 +1,98 @@
+#include "client.hpp"
+
+#include <boost/asio.hpp>
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+using boost::asio::ip::tcp;
+
+namespace meow {
+Client::Client() : io_context() {}
+
+Client::~Client() = default;
+
+void Client::accept_info_about_players() {
+  std::size_t count_of_players;
+  connection >> count_of_players;
+  connection >> my_id;
+
+  for (std::size_t i = 0; i < count_of_players; ++i) {
+    std::string input_msg;
+    while (input_msg.empty()) {
+      std::getline(connection, input_msg);
+    }
+    auto json = json::parse(input_msg);
+    players_info.emplace_back(json);
+  }
+}
+
+void Client::connect(const std::string& host) {
+  size_t delimiter_pos = host.find(':');
+  std::string ip = host.substr(0, delimiter_pos);
+  std::string port = host.substr(delimiter_pos + 1);
+
+  tcp::socket s(io_context);
+  boost::asio::connect(s, tcp::resolver(io_context).resolve(ip, port));
+  connection = tcp::iostream(std::move(s));
+
+  connection << my_name + "\n" << std::flush;
+  accept_info_about_players();
+
+  std::thread([this]() mutable {
+    while (connection) {
+      std::string received_msg;
+      if (!std::getline(connection, received_msg)) {
+        break;
+      }
+      if (received_msg.empty()) continue;
+      auto json = json::parse(received_msg);
+      std::unique_lock l(mtx);
+      if (json.contains("validness")) {
```

Comment:
I am not sure that's a good idea (imagine if you will have another class with the same field --> it will lead to bugs) It would be better to add to json name of a class or come up with a better mechanism.

Reply:
I will try to google it later